src/ua/artcode/udiary/utils/Validator.java

```diff
@@ -1,45 +1,133 @@
+     *               must have not null and not empty email,
+     *               not null and not empty pass.
+     * @throws AppException when user is null, its email is null or empty,
+     * or its pass is null or empty.
+     *
+     * @see User
+     * @see AppException
+     * */
     public static void validateUser(User user) throws AppException {
         if(user == null){
             throw new AppException("empty user");
         }
 
-         /*
-         additional validation...
-         */
+        if (user.getEmail() == null || user.getEmail().isEmpty() ||
+                user.getPass() == null || user.getPass().isEmpty()) {
+            throw new AppException("empty email or pass");
+        }
+    }
+
+    /**
+     * This method verifies user's email in DB (entered as allUsers List).
+     * @param allUsers represents user DB.
+     * @param requestedUser must contain correct email of
+     *                      user, whose data you want to store in DB.
+     * @return true, if there is no user in user DB, which has email equal to requestedUser's.
+     * @throws AppException if params are null.
+     *
+     * @see User
+     * @see AppException
+     * @see List
+     * */
+    public static boolean verifyUserSignIn(List<User> allUsers, User requestedUser) throws AppException {
+        if (allUsers == null || requestedUser == null) {
+            throw new AppException("empty user or list of users");
+        }
+
+        // check if allUsers list contains User with such email as requestedUser's:
+        for (User user: allUsers) {
+            if (user.getEmail().equals(requestedUser.getEmail())) {
+                return false;
+            }
+        }
+
+        // if no user with such email:
+        return true;
     }
 
+    /**
+     * This method verifies user's email and pass in DB.
+     * @param allUsers represents user DB.
+     * @param requestedUser must contain correct email and pass of
+     *                      user, whose data in DB you want to access.
+     * @return reference to user in List, which email and pass are equal to requestedUser's.
+     * @throws AppException if arguments are null or requestedUser's email is wrong
+     * or requestedUser's pass is wrong.
+     *
+     * @see User
+     * @see AppException
+     * @see List
+     * */
+    public static User verifyUserLogIn(List<User> allUsers, User requestedUser) throws AppException {
+        if (allUsers == null || requestedUser == null) {
+            throw new AppException("empty list of users or empty user");
+        }
+
+        // check if allUsers list contains User with such email as requestedUser's:
+        for (User user: allUsers) {
+            if (user.getEmail().equals(requestedUser.getEmail())) {
+                if (user.getPass().equals(requestedUser.getPass())) {
+                    return user;
+                } else {
+                    throw new AppException("wrong password");
```

Comment:
if the first user in allIsers will not fit to the requested user we immediately throw an exception. Is this not strange?

Reply:
By our business logic email must be unique -> when password of User in DB (its email is equal to requested) is not equal to requested, throwing of exception has to pinpoint fact that proposed data (email and pass) is wrong. Or we have to catch it without exceptions?